testing/scalar-functions.test

```diff
@@ -403,6 +403,38 @@ do_execsql_test date-with-invalid-timezone {
   SELECT date('2023-05-18 15:30:45+25:00');
 } {{}}
 
+do_execsql_test typeof-null {
+  SELECT typeof(null);
+} {null}
+
+do_execsql_test typeof-null-case {
+  SELECT typeof(nuLL);
+} {null}
+
+do_execsql_test typeof-text {
+  SELECT typeof('hello');
+} {text}
+
+do_execsql_test typeof-text-empty {
+  SELECT typeof('');
+} {text}
+
+do_execsql_test typeof-integer {
+  SELECT typeof(123);
+} {integer}
+
+do_execsql_test typeof-real {
+  SELECT typeof(1.0);
+} {real}
+
+do_execsql_test typeof-blob {
+  SELECT typeof(x'61');
+} {blob}
+
+do_execsql_test typeof-blob-empty {
+  SELECT typeof(x'');
+} {blob}
```

Comment:
Currently Limbo fails doesn't support blobs very well, and these blob tests fail with an error:

```
thread 'main' panicked at core/translate/expr.rs:1161:38:
not yet implemented
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
    while executing
"exec {*}$command"
    (procedure "evaluate_sql" line 3)
    invoked from within
"evaluate_sql $sqlite_exec $sql"
    (procedure "run_test" line 2)
    invoked from within
"run_test $::sqlite_exec $combined_sql $combined_expected_output"
    (procedure "do_execsql_test" line 5)
    invoked from within
"do_execsql_test typeof-blob {
  SELECT typeof(x'61');
} {blob}"
    (file "./testing/scalar-functions.test" line 430)
    invoked from within
"source $testdir/scalar-functions.test"
    (file "./testing/all.test" line 15)
make: *** [Makefile:59: test-compat] Error 1
```